Document new threading settings (#47575)
dotnet/docs: docs/core/runtime-config/threading.md

| **runtimeconfig.json** | N/A | N/A |
| **Environment variable** | `COMPlus_Thread_AssignCpuGroups` or `DOTNET_Thread_AssignCpuGroups` | `0` - disabled<br/>`1` - enabled |

## Set the size of stack for VM-created threads

> - The default size of stack is dictated by VM and OS policies.

Reviewer: ```suggestion
- The default stack size is dictated by .NET and OS policies. The default stack size for .NET apps is 1.5 MB on Windows and macOS, and 8 MB on Linux. The default stack size when .NET runtime is hosted (for example, COM components) is dictated by the hosting process.
```

> - This setting allows overriding the default.

Reviewer: ```suggestion
- This setting allows overriding the default for threads created by .NET runtime. For example, threads created by the <xref:System.Threading.Thread.%23ctor*> API.
```

> - The default stack size is dictated by .NET and OS policies. The default stack size for .NET apps is 1.5 MB on Windows and macOS, and 8 MB on Linux. The default stack size when .NET runtime is hosted (for example, COM components) is dictated by the hosting process.

Reviewer 2: ```suggestion
- This setting only affects new threads created by .NET. The main thread of the application and threads not created by .NET will be unaffected by this setting.
- The default stack size is dictated by .NET and OS policies. The default stack size for .NET apps is 1.5 MB on Windows and macOS, and 8 MB on Linux. The default stack size when .NET runtime is hosted (for example, COM components) is dictated by the hosting process.
```

> - The specified size should be between 64 kB and 2 GB and a multiple of the natural allocation granularity of the OS.

Author: Looks like the `multiple of the natural allocation granularity of the OS.` is bogus, I don't remember where that came from. Looks like the code will already do the whatever is the necessary adjustment with this number.

| | Setting name | Values | Version introduced |
| - | - | - | - |

> | **runtimeconfig.json** | `System.Threading.DefaultStackSize` | A decimal integer that specifies the stack size, in bytes, for threads created by the .NET runtime | .NET 10 |

Reviewer 3: oh dear 😄
